spring-rabbit-stream/src/test/java/org/springframework/rabbit/stream/listener/RabbitListenerTests.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.rabbit.stream.listener;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.amqp.rabbit.annotation.EnableRabbit;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.rabbit.listener.RabbitListenerContainerFactory;
+import org.springframework.amqp.rabbit.listener.RabbitListenerEndpointRegistry;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.rabbit.stream.config.StreamRabbitListenerContainerFactory;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+import com.rabbitmq.stream.Address;
+import com.rabbitmq.stream.Environment;
+import com.rabbitmq.stream.Message;
+import com.rabbitmq.stream.MessageHandler.Context;
+
+/**
+ * @author Gary Russell
+ * @since 2.4
+ *
+ */
+@SpringJUnitConfig
+public class RabbitListenerTests extends AbstractIntegrationTests {
+
+	@Autowired
+	Config config;
+
+	@Test
+	void simple(@Autowired RabbitTemplate template,
+			@Autowired RabbitListenerEndpointRegistry registry) throws InterruptedException {
+
+		registry.start();
+		template.convertAndSend("test.stream.queue1", "foo");
+		assertThat(this.config.latch1.await(10, TimeUnit.SECONDS)).isTrue();
+		assertThat(this.config.received).isEqualTo("foo");
+	}
+
+	@Test
+	void nativeMsg(@Autowired RabbitTemplate template,
+			@Autowired RabbitListenerEndpointRegistry registry) throws InterruptedException {
+
+		registry.start();
+		template.convertAndSend("test.stream.queue2", "foo");
+		assertThat(this.config.latch2.await(10, TimeUnit.SECONDS)).isTrue();
```

Comment:
You're welcome to steal a [count down latch assertion](https://github.com/rabbitmq/rabbitmq-stream-java-client/blob/33313889c2f1e5e6284d90a3a60a7fc5df867adc/src/test/java/com/rabbitmq/stream/impl/TestUtils.java#L566-L609) from the stream Java client.

With a static method in an utility class:

```java
static CountDownLatchAssert latchAssert(CountDownLatch latch) {
    return new CountDownLatchAssert(latch);
}
```

You end with this in tests:

```java
assertThat(latchAssert(latch)).completes();
```

Reply:
Nice!